src/OrbitGl/CaptureViewElementTest.cpp

```diff
@@ -186,58 +182,20 @@ TEST(CaptureViewElement, MouseWheelEventRecursesToCorrectChildren) {
             container_elem.HandleMouseWheelEvent(kPosOnChild2, kDelta));
 }
 
-class CaptureViewElementRoot : public CaptureViewElement, public ::testing::Test {
- public:
-  CaptureViewElementRoot() : CaptureViewElement(nullptr, &kViewport, &kLayout) {}
-  [[nodiscard]] float GetHeight() const override { return 0; }
-
- private:
-  [[nodiscard]] std::unique_ptr<orbit_accessibility::AccessibleInterface>
-  CreateAccessibleInterface() override {
-    return nullptr;
-  }
-};
-
-TEST_F(CaptureViewElementRoot, RequestUpdateBubblesUpAndIsClearedAfterDrawLoop) {
-  MockBatcher batcher;
-  MockTextRenderer text_renderer;
-  PickingManager picking_manager;
-
-  PrimitiveAssembler primitive_assembler(&batcher, &picking_manager);
-
-  auto simulate_draw_loop_and_check_flags = [&](bool draw, bool update_primitives) {
-    EXPECT_EQ(draw_requested_, draw);
-    EXPECT_EQ(update_primitives_requested_, update_primitives);
-    EXPECT_EQ(HasLayoutChanged(), draw || update_primitives);
-
-    UpdateLayout();
-    if (draw) {
-      Draw(primitive_assembler, text_renderer, CaptureViewElement::DrawContext());
-    }
-    if (update_primitives) {
-      UpdatePrimitives(primitive_assembler, text_renderer, 0, 0, PickingMode::kNone);
-    }
-
-    EXPECT_FALSE(draw_requested_);
-    EXPECT_FALSE(update_primitives_requested_);
-    EXPECT_FALSE(HasLayoutChanged());
-  };
-
-  EXPECT_FALSE(draw_requested_);
-  EXPECT_FALSE(update_primitives_requested_);
-  EXPECT_FALSE(HasLayoutChanged());
-
-  UnitTestCaptureViewContainerElement child(this, &kViewport, &kLayout, 1);
+TEST(CaptureViewElement, RequestUpdateBubblesUpAndIsClearedAfterDrawLoop) {
```

Comment:
Can you test that if there were no changes, both flags are false?

Reply:
`SimulateDrawLoopAndCheckFlags` already checks that they are false after rendering, which is kind of the same as this will be the start state for the next frame. But I've added an explicit check to also execute another round of layouting that should result in no changes.